streams/src/test/java/org/apache/kafka/test/MockApiProcessor.java

```diff
@@ -92,6 +93,20 @@ public void process(final Record<KIn, VIn> record) {
     }
 
     public void checkAndClearProcessResult(final KeyValueTimestamp<?, ?>... expected) {
```

Comment:
Seems we should get rid of this one after we have updated the full DSL to use the new `Record` type.

Reply:
We certainly could, but there's also nothing semantically wrong with making KVT-based assertions, so I figured I'd just leave them both in the test infrastructure.